collectfast/strategies/filesystem.py

```diff
@@ -11,3 +15,27 @@ def get_remote_file_hash(self, prefixed_path: str) -> Optional[str]:
             return self.get_local_file_hash(prefixed_path, self.remote_storage)
         except FileNotFoundError:
             return None
+
+
+class CachingFileSystemStrategy(
+        CachingHashStrategy[FileSystemStorage],
+        FileSystemStrategy,
+):
+    @lru_cache()
+    def get_local_file_hash(self, *args, **kwargs):
+        '''
+        caches the local file hash in memory so the hash is only computed once
+        per run and `post_copy_hook` method the does not duplicate work
+        '''
+        return super().get_local_file_hash(*args, **kwargs)
```

Comment:
I think we can decorate the parent implementation in `base.HashStrategy` instead of overriding and decorating here. I also think we should use `lru_cache(maxsize=None)`.

Reply:
(Feel free to add `maxsize=None` [here](https://github.com/antonagestam/collectfast/blob/master/collectfast/strategies/base.py#L95) as well).

Reply:
> I think we can decorate the parent implementation in `base.HashStrategy` instead of overriding and decorating here. I also think we should use `lru_cache(maxsize=None)`.

I am happy to make that change.  I did it this way to avoiding changing things outside the scope of the issue

Reply:
> (Feel free to add `maxsize=None` [here](https://github.com/antonagestam/collectfast/blob/master/collectfast/strategies/base.py#L95) as well).

I don't know that it will help since we do not operate on each file more than once and the calls are synchronous. I am nervous about an unbounded cache - but probably more out of habit than actual need here.

Reply:
The cache is bounded by the number of static files in the project. We could of course make that configurable but I don't think really that's called for. Having no bound will make more sense than having a bound of 128 here I think.